src/main/kotlin/org/rust/ide/inspections/RsSelfConventionInspection.kt

```diff
@@ -30,9 +32,19 @@ class RsSelfConventionInspection : RsLocalInspectionTool() {
         }
 
     private fun RsFunction.isOwnerCopyable(): Boolean {
-        val implBlock = parentOfType<RsImplItem>() ?: return false
-        val owner = implBlock.typeReference?.type as? TyStructOrEnumBase ?: return false
-        return owner.item.queryAttributes.hasAttributeWithArg("derive", "Copy")
+        val traitOrImpl = parentOfType<RsTraitOrImpl>() ?: return false
```

Comment:
Let's use `RsFunction#owner` here: `parentOfType` might be wrong in cases like this

```
impl X {
  fn foo() { fn bar() { /* no owner here, but there's a parent of type*/ } }
}
```

I *think* that in this particular case it's OK, because we check that the owner is impl or trait, but this relise on some interprocedural reasoning, which makes code harder to read.

Could we switch on `m.owner` directly and fold `copyble` and `owner.isInherentImpl || owner.isTrait` checks into one?

Reply:
owner doesn't use `parentOfType`

Reply:
Look at the wrong place

Reply:
>owner doesn't use parentOfType

Yes, and it's the right thing to do because `parentOfType` might give wrong results. So it's preferable to use `owner` if possible, and I think we can use `owner` here?

Reply:
I try to improve the solution